orca/orca-webhook/src/main/java/com/netflix/spinnaker/orca/webhook/service/WebhookService.java

```diff
@@ -231,17 +241,22 @@ private RestTemplateData getRestTemplateData(
             userConfiguredUrlRestrictions.validateURI(
                 provider.getTargetUrl(destinationUrl, stageData));
 
-        if (!isAllowedRequest(httpMethod, validatedUri)) {
-          String message =
-              String.format(
-                  "http method '%s', uri: '%s' not allowed",
-                  httpMethod.toString(), validatedUri.toString());
-          log.info(message);
-          throw new IllegalArgumentException(message);
+        Optional<WebhookProperties.AllowedRequest> allowedRequest =
+            getAllowedRequest(httpMethod, validatedUri);
+        if (webhookProperties.isAllowedRequestsEnabled()) {
+          if (allowedRequest.isEmpty()) {
```

Comment:
nit: any reason for pulling this logic out of the `isAllowedRequest` method? it reads a bit clunky

Reply:
Yeah, this is clunky.  We need to distinguish between "not allowed" (i.e. not corresponding allow list entry) and "the allowed request feature is disabled".  I could remove the isAllowedRequestsEnabled check from getAllowedRequest but that wouldn't really simplify this code.  Combining the two ifs into one with `&&` would help.  Is that worth it?  Open to suggestions here.

Reply:
Really splitting hairs I guess, but could we move line 244 to line 247, above the second if?